extensions/interactions/EndExploration/directives/oppia-interactive-end-exploration.component.ts

```diff
@@ -20,95 +20,91 @@
+          this.recommendedExplorationIdsWithValue,
+      }
+    ) as EndExplorationCustomizationArgs;
 
-        ctrl.errorMessage = '';
+    const authorRecommendedExplorationIds = recommendedExplorationIds.value;
 
-        if (ctrl.isInEditorPage) {
-          // Display a message if any author-recommended explorations are
-          // invalid.
-          $http.get(EXPLORATION_SUMMARY_DATA_URL_TEMPLATE, {
-            params: {
-              stringified_exp_ids: JSON.stringify(
-                authorRecommendedExplorationIds)
-            }
-          }).then(function(response) {
-            var data = response.data;
-            var foundExpIds = [];
-            data.summaries.map(function(expSummary) {
-              foundExpIds.push(expSummary.id);
-            });
+    this.isInEditorPage = (
+      this.contextService.getPageContext() === (
+        ServicesConstants.PAGE_CONTEXT.EXPLORATION_EDITOR));
+    this.isInEditorPreviewMode = this.isInEditorPage && (
+      this.contextService.getEditorTabContext() === (
+        ServicesConstants.EXPLORATION_EDITOR_TAB_CONTEXT.PREVIEW));
+    this.isInEditorMainTab = this.isInEditorPage && (
+      this.contextService.getEditorTabContext() === (
+        ServicesConstants.EXPLORATION_EDITOR_TAB_CONTEXT.EDITOR));
 
-            var missingExpIds = [];
-            authorRecommendedExplorationIds.forEach(function(expId) {
-              if (foundExpIds.indexOf(expId) === -1) {
-                missingExpIds.push(expId);
-              }
-            });
+    if (this.isInEditorPage) {
+      // Display a message if any author-recommended explorations are
+      // invalid.
+      this.endExplorationBackendApiService.getRecommendExplorationsData(
+        authorRecommendedExplorationIds
+      ).then((response) => {
+        let foundExpIds = [];
+        response.summaries.map((expSummary) => {
+          foundExpIds.push(expSummary.id);
+        });
 
-            if (missingExpIds.length === 0) {
-              ctrl.errorMessage = '';
-            } else {
-              var listOfIds = missingExpIds.join('", "');
-              ctrl.errorMessage = (
-                'Warning: exploration(s) with the IDs "' + listOfIds +
-                '" will not be shown as recommendations because ' +
-                'they either do not exist, or are not publicly viewable.');
-            }
-          });
+        let missingExpIds = [];
+        authorRecommendedExplorationIds.forEach((expId) => {
+          if (foundExpIds.indexOf(expId) === -1) {
+            missingExpIds.push(expId);
+          }
+        });
+
+        if (missingExpIds.length === 0) {
+          this.errorMessage = '';
+        } else {
+          let listOfIds = missingExpIds.join('", "');
+          this.errorMessage = (
+            `Warning: exploration(s) with the IDs "${listOfIds}` +
+            '" will not be shown as recommendations because ' +
+            'they either do not exist, or are not publicly viewable.');
```

Comment:
```suggestion
          this.errorMessage = (
            `Warning: exploration(s) with the IDs "${listOfIds}" `
            'will not be shown as recommendations because '
            'they either do not exist, or are not publicly viewable.'
          );
```

Reply:
Done